devlib/utils/gem5.py

```diff
@@ -38,6 +43,11 @@ def iter_statistics_dump(stats_file):
             res = GEM5STATS_FIELD_REGEX.match(line) 
             if res:
                 k = res.group("key")
-                v = res.group("value").split()
-                cur_dump[k] = v[0] if len(v)==1 else set(v)
+                vtext = res.group("value").split()
```

Comment:
`split()` inside the `map` below, rather than here; so that if/when you log the error in the `except` clause, you will be logging the original value, and not the split list.